### PR TITLE
fix(behavior_velocity_planner): int/uint for path index

### PR DESCRIPTION
**Pull request by soblin**
Signed-off-by: Mamoru Sobue <mamoru.sobue@tier4.jp>

## Description

Fixed overflow of uint.

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/2190

## Tests performed

Fixed error in [scenario test](https://evaluation.tier4.jp/evaluation/oidc/auth/cb/?code=TF_YOOKqleVAgK26TjMdHpv7Cv27MRWYKmSoGKXdDCs.lsipaaGX7wBkrZtWxnky8buMjzqazyUkER1MjIvc1l4&scope=openid+offline_access+t4_profile+token_introspection&state=29ToPXlJqUPR8SojW7Sm4CJcKiNwdXs7)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].
- [X] The PR has been properly tested.
- [X] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.
- [X] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by taikitanaka3** [comment]
@soblin 
I think you also want end index guard  for overflow too.
image

File: planning/behavior_velocity_planner/src/scene_module/intersection/util.cpp [hunk 1]
```diff
@@ -93,7 +93,8 @@ std::optional<std::pair<size_t, size_t>> findLaneIdInterval(
       break;
     }
   }
-  start = std::max<size_t>(0, start - 1);  // the idx of last point before the interval
+  start = static_cast<size_t>(
+    std::max<int>(0, static_cast<int>(start) - 1));  // the idx of last point before the interval
```
Comment (TakaHoribe) [inline planning/behavior_velocity_planner/src/scene_module/intersection/util.cpp:97]: how about this? To get rid of many casting.

```
start = start > 0 ? start - 1 : 0;
```

**Review by TakaHoribe — APPROVED** [review]
Confirmed the issue was solved.

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Pull request merged**